staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -419,3 +566,48 @@ func (p *processorListener) run(stopCh <-chan struct{}) {
 		}
 	}
 }
+
+// shouldResync deterimines if the listener needs a resync. If the listener's resyncPeriod is 0,
+// this always returns false.
+func (p *processorListener) shouldResync(now time.Time) bool {
+	if p.resyncPeriod == 0 {
+		return false
+	}
+
+	return now.After(p.nextResync) || now.Equal(p.nextResync)
+}
+
+// shouldHandleSyncItem returns true if the reflector has started a relisting, or if it has started
+// resyncing and the listener is participating in the current resync.
+func (p *processorListener) shouldHandleSyncItem() bool {
+	return p.relisting || p.resyncing && p.wantSyncItems
```

Comment:
Ok, if I've read your intent correctly, you're doing this so that a listener doesn't listen to half a resync and think its done syncing.  I think the logic may fail on very slow syncs (see my previous comment), but if that's why, what if we considered changing the deltafifo semantics.  What if doing a resync blocked other actions?  The resync then becomes atomic and whether or not we're currently operating a resync becomes fully discoverable at this layer without any plumbing, right?